packages/webpack-plugin/src/plugin.ts

```diff
@@ -0,0 +1,463 @@
+                    const connections = moduleGraph.getOutgoingConnections(module);
+                    for (const connection of connections) {
+                        if (
+                            !isAssetModule(connection.module) &&
+                            isLoadedWithKnownAssetLoader(connection.module)
+                        ) {
+                            connection.setActive(false);
+                        }
+                    }
+                }
+            });
+        }
+
+        /**
+         *  After we have the initial chunks we can calculate the depth and usage of each stylesheet
+         */
+        compilation.hooks.afterChunks.tap({ name: StylableWebpackPlugin.name, stage: 0 }, () => {
+            for (const module of stylableModules) {
+                module.buildMeta.stylable.isUsed = findIfStylableModuleUsed(module, compilation);
+                module.buildMeta.stylable.depth = calcDepth(module, moduleGraph);
+            }
+        });
+
+        compilation.hooks.afterChunks.tap(StylableWebpackPlugin.name, () => {
+            const optimizer = this.stylable.optimizer!;
+            const optimizeOptions = this.options.optimize;
+            const sortedModules = getSortedModules(stylableModules);
+
+            const {
+                usageMapping,
+                namespaceMapping,
+                namespaceToFileMapping,
+            } = createOptimizationMapping(sortedModules, optimizer);
+
+            if (!this.options.unsafeMuteDiagnostics.DUPLICATE_MODULE_NAMESPACE) {
+                reportNamespaceCollision(namespaceToFileMapping, compilation.errors);
+            }
+
+            for (const module of sortedModules) {
+                const buildMeta = getStylableBuildMeta(module);
+                const { css, exports, globals } = buildMeta;
+
+                const ast = parse(css);
+
+                optimizer.optimizeAst(
+                    optimizeOptions,
+                    ast,
+                    usageMapping,
+                    this.stylable.delimiter,
+                    exports,
+                    globals
+                );
+
+                buildMeta.css = optimizeOptions.minify
+                    ? optimizer.minifyCSS(ast.toString())
+                    : ast.toString();
+
+                if (optimizeOptions.shortNamespaces) {
+                    buildMeta.namespace = namespaceMapping[buildMeta.namespace];
+                }
+            }
+        });
+    }
+    private chunksIntegration(
+        compilation: Compilation,
+        staticPublicPath: string,
+        stylableModules: Set<NormalModule>,
+        assetsModules: Map<string, NormalModule>
+    ) {
+        const { runtimeTemplate } = compilation;
+
+        /**
+         * As a work around unknown behavior
+         * if this plugin will run inside a child compilation we do not emit css assets
+         */
+        if (!compilation.compiler.isChild()) {
```

Comment:
this if can be checked where .chunksIntegration() is called, making the entire method less indented

Reply:
it looks like that but the intention will be wrong. for now we don't have more stuff to do in the chunksIntegration but when new concern will arise it should consider whether to work in a child compiler.
Also I wanted to make the main flow branchless